### PR TITLE
Fix known issue of job context using.

### PR DESCRIPTION
**Pull request by BalaBalaYi**
### What changes were proposed in this pull request?

1. Rename the function in ```JobContext``` if the method can read and write the core 'nodes' object.
2. Other fixed.

### Why are the changes needed?

To fix known issue.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

UT and training with failover(doing).

### CODE REVIEW

File: dlrover/python/master/node/job_context.py [hunk 1]
```diff
@@ -48,15 +48,13 @@ def next_action(
     ):
         return self._action_queue.next_action(instance=instance)
 
-    @property
-    def ps_nodes(self) -> Dict[int, Node]:
+    def get_and_update_ps_nodes(self) -> Dict[int, Node]:
```
Comment (workingloong) [inline dlrover/python/master/node/job_context.py:51]: There is no any code to update `job_nodes["PS"]`

Comment (samplise) [reply dlrover/python/master/node/job_context.py:51]: `get_and_update_ps_nodes` -> `get_mutable_ps_nodes`

Comment (BalaBalaYi) [reply dlrover/python/master/node/job_context.py:51]: ```get_mutable_ps_nodes ``` lgtm

**Review by samplise — APPROVED** [review]
lgtm

**Pull request merged**